modules/social_features/social_follow_taxonomy/social_follow_taxonomy.module

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * @file
+ * Contains social_follow_taxonomy.module.
+ */
+
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Url;
+use Drupal\message\Entity\Message;
+use Drupal\node\NodeInterface;
+use Drupal\taxonomy\TermInterface;
+
+/**
+ * Implements hook_theme().
+ */
+function social_follow_taxonomy_theme($existing, $type, $theme, $path) {
+  return [
+    'flag__follow_term' => [
+      'base hook' => 'flag',
+    ],
+    'activity__node__followed' => [
+      'base hook' => 'activity',
+    ],
+  ];
+}
+
+/**
+ * Implements hook_social_user_account_header_account_links().
+ *
+ * Adds the "Following tags" link to the user menu.
+ */
+function social_follow_taxonomy_social_user_account_header_account_links(array $context) {
+  return [
+    'my_tags' => [
+      '#type' => 'link',
+      '#attributes' => [
+        'title' => new TranslatableMarkup("View tags I'm following"),
+      ],
+      '#title' => new TranslatableMarkup('Following tags'),
+      '#weight' => 1001,
+    ] + Url::fromRoute('view.following_tags.following_tags')->toRenderArray(),
+  ];
+}
+
+/**
+ * Implements hook_page_attachments().
+ */
+function social_follow_taxonomy_page_attachments(array &$page) {
+  $page['#attached']['library'][] = 'social_follow_taxonomy/social_follow_taxonomy';
+}
+
+/**
+ * Implements hook_theme_suggestions_alter().
+ */
+function social_follow_taxonomy_theme_suggestions_activity(array $variables) {
+  $suggestions = [];
+
+  $activity = $variables['elements']['#activity'];
+  $message = Message::load($activity->field_activity_message->target_id);
+
+  if ($message->getTemplate()->id() == 'create_node_following_tag') {
+    $entity_type = $activity->field_activity_entity->first()->target_type;
+
+    if ($entity_type === 'node') {
+      $suggestions[] = 'activity__' . $entity_type . '__followed';
+    }
+  }
+
+  return $suggestions;
+}
+
+/**
+ * Extends variables for activity template.
+ *
+ * Implements hook_preprocess_activity().
+ * {@inheritdoc}
+ */
+function social_follow_taxonomy_preprocess_activity(&$variables) {
+  $activity = $variables['elements']['#activity'];
+
+  $message = Message::load($activity->field_activity_message->target_id);
+  $message_template_id = $message->getTemplate()->id();
+
+  if ($message_template_id == 'create_node_following_tag') {
```

Comment:
@VolodymyrMartsynenko Could you give more comments about what happens here?

Reply:
I added an additional comments for this.